src/services/item/plugins/html/service.ts

```diff
@@ -0,0 +1,196 @@
+import extract from 'extract-zip';
+import fs from 'fs';
+import { lstat, readdir } from 'fs/promises';
+import { mkdir } from 'fs/promises';
+import mime from 'mime';
+import path from 'path';
+import { pipeline } from 'stream/promises';
+import tmp from 'tmp-promise';
+import { v4 } from 'uuid';
+
+import { MultipartFile } from '@fastify/multipart';
+import { FastifyBaseLogger } from 'fastify';
+
+import { TMP_FOLDER } from '../../../../utils/config';
+import { UnauthorizedMember } from '../../../../utils/errors';
+import { Repositories } from '../../../../utils/repositories';
+import FileService from '../../../file/service';
+import { Actor, Member } from '../../../member/entities/member';
+import { Item } from '../../entities/Item';
+import { GraaspHtmlError, HtmlImportError, HtmlInvalidFileError } from './errors';
+import { DEFAULT_MIME_TYPE } from './h5p/constants';
+import { HtmlValidator } from './validator';
+
+/**
+ * Implementation for the Html service
+ */
+export abstract class HtmlService {
+  protected readonly fileService: FileService;
+  protected readonly pathPrefix: string;
+  protected readonly validator: HtmlValidator;
+  protected readonly mimetype: string;
+  protected readonly extension: string;
+
+  protected readonly tempDir: string;
+
+  constructor(
+    fileService: FileService,
```

Comment:
Maybe this parent class should initialize given the file type?

Reply:
Interesting, but then we need to pass the `fileStorage.config`, etc?